ui/static/ui/js/listing_resources.jsx

```diff
@@ -240,6 +240,40 @@ define('listing_resources', ['react', 'jquery', 'lodash', 'utils'],
     });
 
     var ListingResource = React.createClass({
+      wrapDescription: function(resource) {
+        var description = "No description provided.";
+        if (resource.description !== "") {
+          if (resource.description.length > 120) {
+            var wrapDesc ;
+            var positionOfSpace = resource.description.indexOf(" ",  110);
+
+            if (positionOfSpace > 0) {
+              wrapDesc = resource.description.substring(0, positionOfSpace);
+            } else {
+              wrapDesc = resource.description.substring(0, 120);
+            }
+
+            description = <span>
+              {wrapDesc} <br/>
+              <a onClick={this.expandDescriptionClick}
+              className="link-description-more-less">
+               Read more <i className="fa fa-chevron-down"></i>
+              </a>
+            </span>;
+          } else {
+            description = resource.description;
+          }
+        }
+        this.setState({description: description});
+      },
+      componentWillMount: function() {
+        var resource = this.props.resource;
+        this.wrapDescription(resource);
```

Comment:
In general we should compute values for render in the `render` function, there is usually not a need to precompute this information and store it in `state`. See [What shouldn't go in state](https://facebook.github.io/react/docs/interactivity-and-dynamic-uis.html#what-shouldnt-go-in-state)